plugins/modules/cloudfront_distribution.py

```diff
@@ -1685,10 +1685,16 @@ def validate_origins(self, client, config, origins, default_origin_domain_name,
             self.module.fail_json_aws(e, msg="Error validating distribution origins")
 
     def validate_s3_origin_configuration(self, client, existing_config, origin):
-        if origin['s3_origin_access_identity_enabled'] and existing_config.get('s3_origin_config', {}).get('origin_access_identity'):
-            return existing_config['s3_origin_config']['origin_access_identity']
         if not origin['s3_origin_access_identity_enabled']:
             return None
+
+        if origin['s3_origin_access_identity_enabled']:
```

Comment:
We've already returned None by this point, if s3_origin_access_identity_enabled isn't defined.  I think we can forego the extra level of nesting.